src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,66 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String COMMA = ",";
+    private static final String BUY = "buy";
+    private static final String SUPPLY = "supply";
+    private static final String RESULT = "result";
+    private static final int OPERATION_TYPE_INDEX = 0;
+    private static final int OPERATION_VALUE_INDEX = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] dataFromFile = readFromFile(fromFileName).split(System.lineSeparator());
+        writeToFile(toFileName, createReport(dataFromFile));
+    }
+
+    private String createReport(String[] dataFromFile) {
+        int buy = 0;
+        int supply = 0;
+        StringBuilder builder = new StringBuilder();
+        for (String line : dataFromFile) {
+            String[] records = line.split(COMMA);
+            if (records[OPERATION_TYPE_INDEX].equals(BUY)) {
+                buy += Integer.valueOf(records[OPERATION_VALUE_INDEX]);
+            }
+            if (records[OPERATION_TYPE_INDEX].equals(SUPPLY)) {
+                supply += Integer.valueOf(records[OPERATION_VALUE_INDEX]);
+            }
+        }
+        int result;
+        result = supply - buy;
+        return builder.append(SUPPLY + COMMA).append(supply).append(System.lineSeparator())
+                .append(BUY + COMMA).append(buy).append(System.lineSeparator())
+                .append(RESULT + COMMA).append(result).toString();
+    }
+
+    private String readFromFile(String fromFileName) {
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            StringBuilder builder = new StringBuilder();
+            int value = bufferedReader.read();
+            while (value != -1) {
+                builder.append((char) value);
+                value = bufferedReader.read();
+            }
+            return builder.toString();
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("Can't find file with name " + fromFileName, e);
```

Comment:
I think only one catch will be enough, if FileNotFoundException occurs all details will be in 'e'